template/CHART_NAME/templates/daemonset.yaml

```diff
@@ -8,34 +8,26 @@ kind: DaemonSet
 metadata:
   name: {{ template "common.names.fullname" . }}
   namespace: {{ include "common.names.namespace" . | quote }}
-  labels: {{- include "common.labels.standard" . | nindent 4 }}
+  labels: {{- include "common.labels.standard" ( dict "customLabels" .Values.commonLabels "context" $ ) | nindent 4 }}
     app.kubernetes.io/component: %%COMPONENT_NAME%%
-    {{- if .Values.commonLabels }}
-    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
-    {{- end }}
   {{- if .Values.commonAnnotations }}
   annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
   {{- end }}
 spec:
   {{- if .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy }}
   updateStrategy: {{- toYaml .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy | nindent 4 }}
   {{- end }}
+  {{- $podLabels := merge .Values.%%MAIN_OBJECT_BLOCK%%.podLabels .Values.commonLabels }}
   selector:
-    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+    matchLabels: {{- include "common.labels.matchLabels" ( dict "customLabels" $podLabels "context" $ ) | nindent 6 }}
```

Comment:
I use podLabels for networkpolicies [example](https://github.com/bitnami/charts/blob/main/bitnami/redis/templates/networkpolicy.yaml#L57) too, and I don't want them added to matchLabels

Reply:
also this make all charts backward incompatible as selector field is immutable.

Reply:
That's the reason why `common.labels.matchLabels` must ignore any custom label and exclusively focus on changes done to `app.kubernetes.io/name` or `app.kubernetes.io/instance`

Reply:
then the matchLabels snippet should be rewriten with
```
{{ pick (
    merge
        (include "common.tplvalues.render" (dict "value" $customLabels "context" .context) | fromYaml)
        (dict
            "app.kubernetes.io/name" (include "common.names.name" .context)
            "app.kubernetes.io/instance" .context.Release.Name
        )
    ) "app.kubernetes.io/name" "app.kubernetes.io/instance"
    | toYaml
}}
```

ref: [pick](http://masterminds.github.io/sprig/dicts.html#pick) to keep only selected keys